codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/http/ResponseBindingGenerator.kt

```diff
@@ -232,10 +232,17 @@ class ResponseBindingGenerator(protocolConfig: ProtocolConfig, private val opera
                 timestampFormatType
             )
         } else {
-            rust(
-                "let $parsedValue: Vec<${coreType.render(true)}> = #T::read_many(headers)?;",
-                headerUtil
-            )
+            if (coreType == RustType.String) {
```

Comment:
thanks for getting to the bottom of this issue! This fix isn't quite right—this doesn't work for the case where we need to split a string list on strings. I think that we actually need `if (rustType == RustType.String)` before potentially stripping off the container. 

I'm happy to take this over if you prefer as it'd be great to land it prior to Tuesday's release. We'll also need to add a protocol test validating this behavior.